src/installer.js

```diff
@@ -21,261 +21,231 @@ function defaultRename (dest, src) {
-    args.push('--no-msi')
+  /**
+   * Sync remote releases.
+   */
+  syncRemoteReleases () {
+    if (!this.options.remoteReleases) {
+      return
+    }
+
+    this.options.logger(`Syncing package at ${this.stagingDir}`)
+
+    const cmd = path.join(this.vendorDir, 'squirrel', 'SyncReleases.exe')
+    const args = [
+      '--url',
+      this.options.remoteReleases,
+      '--releaseDir',
+      this.squirrelDir
+    ]
+
+    return fs.ensureDir(this.squirrelDir, '0755')
+      .then(() => spawn(cmd, args, this.options.logger))
+      .catch(common.wrapError('syncing remote releases'))
   }
-
-  return spawn(cmd, args, options.logger)
-    .then(() => dir)
-    .catch(common.wrapError('releasifying package'))
-}
-
-/**
- * Move the package files to the specified destination.
- */
-function movePackage (options, dir) {
-  const packagePattern = path.join(dir, 'squirrel', '*')
-
-  common.movePackage(packagePattern, options, dir)
 }
 
 /* ************************************************************************** */
 
-module.exports = function (data, callback) {
+module.exports = (data, callback) => {
   data.rename = data.rename || defaultRename
   data.logger = data.logger || defaultLogger
 
-  let options
-
-  const promise = getDefaults(data)
-    .then(defaults => getOptions(data, defaults))
-    .then(generatedOptions => {
-      options = generatedOptions
-      return data.logger(`Creating package with options\n${JSON.stringify(options, null, 2)}`)
-    }).then(() => common.createDir(options))
-    .then(dir => createSubdirs(options, dir))
-    .then(dir => createContents(options, dir))
-    .then(dir => createPackage(options, dir))
-    .then(dir => findPackage(options, dir))
-    .then(data => syncRemoteReleases(options, data.dir, data.pkg))
-    .then(data => releasifyPackage(options, data.dir, data.pkg))
-    .then(dir => movePackage(options, dir))
+  if (callback) {
+    console.warn('The node-style callback is deprecated. In a future major version, it will be' +
+                 'removed in favor of a Promise-based async style.')
+  }
+
+  const installer = new SquirrelInstaller(data)
+
+  const promise = installer.generateDefaults()
+    .then(() => installer.generateOptions())
+    .then(() => data.logger(`Creating package with options\n${JSON.stringify(installer.options, null, 2)}`))
+    .then(() => installer.createStagingDir())
+    .then(() => installer.createContents())
+    .then(() => installer.createPackage())
+    .then(() => installer.syncRemoteReleases())
+    .then(() => installer.releasifyPackage())
+    .then(() => installer.movePackage())
     .then(() => {
-      data.logger(`Successfully created package at ${options.dest}`)
-      return options
+      data.logger(`Successfully created package at ${installer.options.dest}`)
+      return installer.options
     }).catch(err => {
       data.logger(common.errorMessage('creating package', err))
       throw err
     })
 
   return nodeify(promise, callback)
```

Comment:
As with `-redhat` we also need to remove callbacks from here.

Reply:
We should do that in a separate PR (in both modules).

Reply:
Ok, sounds good